src/rgw/rgw_trim_bilog.cc

```diff
@@ -571,13 +729,25 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
       return set_cr_error(retcode);
     }
 
+    retcode = maybe_remove_generation();
+    if (retcode < 0) {
+      ldpp_dout(dpp, 4) << "remove removing old generation from log: "
+			<< cpp_strerror(retcode) << dendl;
+      return set_cr_error(retcode);
+    }
+    if (toclean) {
+      yield call(new BucketCleanIndexCollectCR(dpp, store, *new_bi,
+					       *toclean));
```

Comment:
nit: derefs `new_bi` without checking for empty

Reply:
Fixed